tools/scene-viewer/src/main.rs

```diff
@@ -0,0 +1,233 @@
+use bevy::{asset::AssetServerSettings, input::mouse::MouseMotion, prelude::*};
+
+fn main() {
+    println!(
+        "
+Controls:
+    WSAD   - forward/back/strafe left/right
+    LShift - 'run'
+    E      - up
+    Q      - down
+    L      - animate light direction
+    U      - toggle shadows
+    5/6    - decrease/increase shadow projection width
+    7/8    - decrease/increase shadow projection height
+    9/0    - decrease/increase shadow projection near/far
+"
+    );
+    App::new()
+        .insert_resource(AmbientLight {
+            color: Color::WHITE,
+            brightness: 1.0 / 5.0f32,
+        })
+        .insert_resource(AssetServerSettings {
+            asset_folder: "../../".to_string(),
+            watch_for_changes: true,
+        })
+        .insert_resource(WindowDescriptor {
+            title: "bevy scene viewer".to_string(),
+            ..default()
+        })
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(update_lights)
+        .add_system(camera_controller)
+        .run();
+}
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    let scene_path = std::env::args()
+        .nth(1)
+        .unwrap_or_else(|| "assets/models/FlightHelmet/FlightHelmet.gltf#Scene0".to_string());
+    commands.spawn_scene(asset_server.load(&scene_path));
+    commands
+        .spawn_bundle(PerspectiveCameraBundle {
+            transform: Transform::from_xyz(0.0, 0.0, 1.0).looking_at(Vec3::ZERO, Vec3::Y),
+            ..default()
+        })
+        .insert(CameraController::default());
+    const HALF_SIZE: f32 = 1.0;
+    commands.spawn_bundle(DirectionalLightBundle {
+        directional_light: DirectionalLight {
+            shadow_projection: OrthographicProjection {
+                left: -HALF_SIZE,
+                right: HALF_SIZE,
+                bottom: -HALF_SIZE,
+                top: HALF_SIZE,
+                near: -10.0 * HALF_SIZE,
+                far: 10.0 * HALF_SIZE,
+                ..default()
+            },
+            shadows_enabled: false,
+            ..default()
+        },
+        ..default()
+    });
```

Comment:
we should not spawn a directional light if there's one in the scene

Reply:
I've removed this, but I want to add back auto-spawning of a directional light if there is no light and so I won't resolve this thread yet, as a reminder.

Reply:
Done.